tools/ModelGeneration/Mathematica/N2HDM.m

```diff
@@ -0,0 +1,408 @@
+(* ::Package:: *)
+
+(* ::Chapter:: *)
+(*In this section the scalar potential is defined. As an example the N2HDM is shown*)
+
+
+(*Define higgs fields*)
+higgsbase = {\[Rho]1, \[Rho]2, \[Eta]1, \[Eta]2, \[Psi]1, \[Psi]2, \[Zeta]1, \[Zeta]2, \[Zeta]S};
+
+
+(*Assign vevs at T=0*)
+higgsvev = {0, 0, 0, 0, 0, 0, v1, v2, vs};
+
+
+(*Assign vevs at T != 0*)
+higgsvevFiniteTemp = {0, wcb, 0, 0, 0, wcp, w1, w2, ws};
+
+
+(*Replacement list for the vevs*)
+VEVRep = Table[i[[1]]->i[[2]],{i,Transpose[{higgsbase,higgsvev}]}];
+
+
+(*Replacement list set fields zero*)
+RepHiggsZero = Table[i->0,{i,higgsbase}];
+
+
+(*Define number of Higgses*)
+nHiggs = Length[higgsbase];
+
+
+(*Define input parameters of the potential*)
+InputParameters={tanbeta, vs, L1, L2, L3, L4, L5, L6, L7, L8,m12Sq};
+
+
+(*Define parameters of the potential*)
+par = {m11Sq, m22Sq, m12Sq, L1, L2, L3, L4, L5, msSq, L6, L7, L8};
+
+
+(*Define doublets and scalar fields*)
+Phi1 = {higgsbase[[1]] + higgsbase[[3]]*I, higgsbase[[7]] + higgsbase[[5]]*I}/Sqrt[2];
+Phi2 = {higgsbase[[2]] + higgsbase[[4]]*I, higgsbase[[8]] + higgsbase[[6]]*I}/Sqrt[2];
+S = higgsbase[[9]];
+(*Print fields*)
+Phi1//MatrixForm//TraditionalForm
+Phi2//MatrixForm//TraditionalForm
+S//MatrixForm//TraditionalForm
+
+
+(*Write the potential*)
+$Assumptions = higgsbase \[Element] Reals;
+C11 = Simplify[ConjugateTranspose[Phi1] . Phi1];
+C12 = Simplify[ConjugateTranspose[Phi1] . Phi2];
+C21 = Simplify[ConjugateTranspose[Phi2] . Phi1];
+C22 = Simplify[ConjugateTranspose[Phi2] . Phi2];
+CS = Simplify[Conjugate[S]*S];
+VHiggs = par[[1]]*C11 + par[[2]]*C22 - par[[3]]*(C12 + C21) + par[[4]]/2*C11^2 + par[[5]]/2*C22^2 + par[[6]]*C11*C22 + par[[7]]*C12*C21 + par[[8]]/2*(C12^2 + C21^2) + par[[9]]/2*CS + par[[10]]/8*CS^2 + par[[11]]/2*C11*CS + par[[12]]/2*C22*CS//Simplify
+
+
+(*Calculate Tadpoles for Minimisation conditions*)
+VHiggsGrad=D[VHiggs,{higgsbase}]/.VEVRep//Transpose//Simplify;
+{"Derivative with respective to" higgsbase,VHiggsGrad}//Transpose//TableForm
+
+
+ListOfDependentParameter = Select[RemoveDuplicates[Join[RemoveDuplicates[InputParameters],RemoveDuplicates[Join[par,higgsvev]]]],MemberQ[Join[par,higgsvev],#]&];
```

Comment:
Before this line, you need to load import the helper functions, otherwise `RemoveDuplicates` remains undefined.

Reply:
Done.